### PR TITLE
Fixing bonnie test

### PR DESCRIPTION
**Pull request by Naresh-ibm**
Tool creates and deletes sequential and random files in a dir

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
The previous script was not running in all the systems so I have added a option parameter "data_size_to_pass" (-s with 0) to the command that script uses to run the bonnie.  the script was giving error of "no space left on disk" and fails. After adding the the option -s0 the script runs without any issue.
attached the log files with and without parameter for your reference.

[bonnie-fail-job.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/663025/bonnie-fail-job.txt)
[pass-with-paramter-bonnie.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/663029/pass-with-paramter-bonnie.txt)

File: perf/bonnie.py [hunk 1]
```diff
@@ -50,21 +56,25 @@ def setUp(self):
...
         args = []
         args.append('-d %s' % scratch_dir)
         args.append('-n %s' % number_to_stat)
+        args.append('-s %s' % data_size_to_pass)
         if uid_to_use is not None:
             args.append('-u %s' % uid_to_use)
 
         cmd = ('%s/bonnie++ %s' % (self.srcdir, " ".join(args)))
-        process.run(cmd)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
```
Comment (balamuruhans) [inline perf/bonnie.py]: `if process.system(cmd, shell=True, ignore_status=True):` will be sufficient

File: perf/bonnie.py [hunk 2]
```diff
@@ -50,21 +56,25 @@ def setUp(self):
...
+            self.fail("test failed")
```
Comment (balamuruhans) [inline perf/bonnie.py]: Better debug message can be used.

File: perf/bonnie.py.data/bonnie.yaml [hunk 3]
```diff
@@ -8,6 +8,7 @@
 
 # Valid options in avocado test are bellow:
```
Comment (balamuruhans) [inline perf/bonnie.py.data/bonnie.yaml]: %s/bellow/below

**Comment by narasimhan-v** [comment]
Make the commit message more relevant.

Ex:
Fixing bonnie test

Added -s parameter to fix the "no space left on disk"
Changed the bonnie tool version to latest.

File: perf/bonnie.py.data/Readme.txt [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+Example : ./bonnie++ -u root -d /mnt -s0 -b -n 10:100:10:1000
+
+so one has to pass following parameter in the yaml file
+
+scratch-dir: /mnt              (disk mounted on some dir. say /mnt here)
+uid-to-use: root               (user name or it UUID, here it is name i,e root)
+number-to-stat: 10:0:0:2:8192  (Number of files to create file test)
+size_to_pass: 0                (dataset size, here we are passing 0 to skip it as we running file related test)
+
```
Comment (PraveenPenguin) [inline perf/bonnie.py.data/Readme.txt]: some white spaces in README please remove this

File: perf/bonnie.py [hunk 5]
```diff
@@ -36,12 +40,14 @@ class Bonnie(Test):
...
+
         """
         Build bonnie++
         Source:
-         http://www.coker.com.au/bonnie++/experimental/bonnie++-1.96.tgz
+         http://www.coker.com.au/bonnie++/bonnie++-1.03e.tgz
         """
-        tarball = self.fetch_asset('http://www.coker.com.au/bonnie++/experimental/bonnie++-1.96.tgz')
+        tarball = self.fetch_asset('http://www.coker.com.au/\
+                                    bonnie++/bonnie++-1.03e.tgz')
```
Comment (PraveenPenguin) [inline perf/bonnie.py]: please use expire param in` fetch_asset` as it help to download again it if expire time got expired .

File: perf/bonnie.py [hunk 6]
```diff
@@ -50,21 +56,25 @@ def setUp(self):
...
 
     def test(self):
+
         """
         Run 'bonnie' with its arguments
         """
         scratch_dir = self.params.get('scratch-dir', default=self.srcdir)
         uid_to_use = self.params.get('uid-to-use', default=None)
         number_to_stat = self.params.get('number-to-stat', default=2048)
+        data_size_to_pass = self.params.get('data_size_to_pass', default=0)
```
Comment (PraveenPenguin) [inline perf/bonnie.py]: as you are extending this can all these definition move to Setup fn ?

**Comment by PraveenPenguin** [comment]
@Naresh-ibm i think still review comment not addressed in rebase can you please have a look , also please attach job log

**Comment by Naresh-ibm** [comment]
[edited-pass-bonnie.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/664105/edited-pass-bonnie.txt)

Changed the respective files. please review and let me know further if needed anything.
added the log file after edition.

**Comment by narasimhan-v** [comment]
Looks good to me

File: perf/bonnie.py [hunk 7]
```diff
@@ -36,12 +38,20 @@ class Bonnie(Test):
...
+         http://www.coker.com.au/bonnie++/experimental/bonnie++-1.03e.tgz
         """
-        tarball = self.fetch_asset('http://www.coker.com.au/bonnie++/experimental/bonnie++-1.96.tgz')
+
+        self.scratch_dir = self.params.get('scratch-dir', default=self.srcdir)
+        self.uid_to_use = self.params.get('uid-to-use', default=None)
+        self.number_to_stat = self.params.get('number-to-stat', default=2048)
+        self.data_size = self.params.get('data_size_to_pass', default=0)
+
+        tarball = self.fetch_asset('http://www.coker.com.au/bonnie++/'
```
Comment (PraveenPenguin) [inline perf/bonnie.py]: please use expire param in fetch_asset

File: perf/bonnie.py.data/Readme.txt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Bonnie++ is a disk and file system benchmarking tool for measuring I/O performance. With Bonnie++ you can quickly and easily produce a meaningful value to represent your current file system performance.
+Example : ./bonnie++ -u root -d /mnt -s0 -b -n 10:100:10:1000
+so one has to pass following parameter in the yaml file
+scratch-dir: /mnt (disk mounted on some dir. say /mnt here)
+uid-to-use: root (user name or it UUID, here it is name i,e root)
+number-to-stat: 10:0:0:2:8192 (Number of files to create file test)
+size_to_pass: 0 (dataset size, here we are passing 0 to skip it as we running file related test)
+
```
Comment (PraveenPenguin) [inline perf/bonnie.py.data/Readme.txt]: please remove this white space

File: perf/bonnie.py [hunk 9]
```diff
@@ -50,21 +60,20 @@ def setUp(self):
...
-
         args = []
-        args.append('-d %s' % scratch_dir)
-        args.append('-n %s' % number_to_stat)
-        if uid_to_use is not None:
-            args.append('-u %s' % uid_to_use)
+        args.append('-d %s' % self.scratch_dir)
+        args.append('-n %s' % self.number_to_stat)
+        args.append('-s %s' % self.data_size)
+        if self.uid_to_use is not None:
```
Comment (balamuruhans) [inline perf/bonnie.py]: `if self.uid_to_use:` suffice the condition

File: perf/bonnie.py (hunk 7 above)
Comment (Naresh-ibm) [inline perf/bonnie.py]: done. and did a git push with changes.

File: perf/bonnie.py (hunk 9 above)
Comment (Naresh-ibm) [inline perf/bonnie.py]: Thanks Bala, Changed it.

File: perf/bonnie.py.data/Readme.txt (hunk 8 above)
Comment (Naresh-ibm) [inline perf/bonnie.py.data/Readme.txt]: done. removed the the white space i,e an empty line.

**Comment by balamuruhans** [comment]
LGTM

**Comment by PraveenPenguin** [comment]
@narasimhan-v @balamuruhans thanks

**Comment by PraveenPenguin** [comment]
@Naresh-ibm thanks looks good :)

**Comment by Naresh-ibm** [comment]
@narasimhan-v @balamuruhans @PraveenPenguin  Thank you so much for all your quick responses and reviews. Leant many things in my first commit. Thank you all once again :-)

**Comment by narasimhan-v** [comment]
@Naresh-ibm welcome to Upstreaming. Looking forward to many more commits from you !

**Pull request merged**